pkg/controller/dnszone/awsactuator.go

```diff
@@ -227,6 +227,12 @@ func (a *AWSActuator) Refresh() error {
 		}
 		logger.Debug("Found hosted zone")
 		a.zoneID = &zoneID
+
+		// Update dnsZone status now that we have the zoneID
+		if err := a.ModifyStatus(); err != nil {
```

Comment:
How would you feel about a different approach where the `DeleteAWSRecordSets` gets passed the zone ID and zone name rather than the `DNSZone`?

Reply:
And if we are going to set the DNSZone status in the `Refresh` function, then we should do that for all of the platforms.

Reply:
That works as well (how it previously worked). It just seemed odd that we pulled the zoneID from AWS and didn't update the DNSZone object status to reflect the new data.

Is there any reason we shouldn't be updating the DNSZone status as soon as possible?

Reply:
It seems fine to me that the information is stored in the status of the DNSZone rather than the actuator. If we are going to do that, we should just get rid of the `ModifyStatus` function. Maybe @twiest has some insight into this?

Reply:
With the currently proposed behavior in this PR, we can make ModifyStatus an internal function for the actuators that runs at the end of a Refresh()

Reply:
There is a difference in behavior between when `Refresh()` fails and when `ModifyStatus()` fails (`ModifyStatus()` sets the `RequeueAfter` value).

I'm not sure how significant this is, but just something to look into.

There was a reason that we did `ModifyStatus()` separate, but I don't remember it now.

I looked through the code and it seems like it would act the same, so maybe whatever reason we had has now been refactored out of the code.

I'm a fan of having the actuator interface have as few methods as necessary, so this change seems fine assuming it won't break something. :)